src/boot/bootctl-install.c

```diff
@@ -830,7 +1087,19 @@ int verb_install(int argc, char *argv[], void *userdata) {
                                 return r;
                 }
 
-                r = install_binaries(arg_esp_path, arch, install);
+                r = install_binaries_extra(
+                                arg_esp_path,
+                                arch,
+                                install,
+                                &default_loader_path,
+                                &default_loader_description);
+                if (r == 0)
+                        r = install_binaries(
+                                        arg_esp_path,
+                                        arch,
+                                        install,
+                                        &default_loader_path,
+                                        &default_loader_description);
                 if (r < 0)
```

Comment:
`install_binaries_extra()` returns 1 if it did anything. In that case, we'll skip the call to `install_binaries()`. That seems … unexpected. Assuming that's an error, please use the usual pattern instead with
```C
r = foo()
if (r < 0)
    …;
r = bar();
if (r < 0)
   ...;
```
#